src/helpers.js

```diff
@@ -52,71 +53,42 @@ function mergeHashes(source, target = {}, old, keepRemoved = false) {
         old[key],
         keepRemoved
       )
-      target[key] = nested.new
       if (Object.keys(nested.old).length) {
         old[key] = nested.old
       }
     }
-    else if (target[key] !== undefined) {
-      if (typeof source[key] === 'string' || Array.isArray(source[key])) {
-        target[key] = source[key]
-      }
-      else {
-        old[key] = source[key]
-      }
-    }
     else {
-      // support for plural in keys
-      const pluralMatch = /_plural(_\d+)?$/.test(key)
-      const singularKey = key.replace(/_plural(_\d+)?$/, '')
-
-      // support for context in keys
-      const contextMatch = /_([^_]+)?$/.test(singularKey)
-      const rawKey = singularKey.replace(/_([^_]+)?$/, '')
-
-      if (
-        (contextMatch && target[rawKey] !== undefined) ||
-        (pluralMatch && target[singularKey] !== undefined)
-      ) {
-        target[key] = source[key]
-      }
-      else if (keepRemoved) {
-        target[key] = source[key]
-        old[key] = source[key]
+      let mergeTarget
+      if (target[key] !== undefined) {
+        mergeTarget = (typeof source[key] === 'string' || Array.isArray(source[key]))
```

Comment:
I'd prefer without the wrapping parentheses here

Reply:
Ok.